src/frontend/utils/conversejs/converse-plugins/chatPlugin.ts

```diff
@@ -35,37 +39,90 @@ const addChatPlugin = (xmpp: XMPP) =>
             switch (stanza.nodeName) {
               case StanzaType.MESSAGE:
                 switch (determineMessageType(stanza)) {
-                  case StanzaMessageType.CLASSIC_MESSAGE:
-                    const msg: ChatMessageType = {
-                      sentAt: DateTime.now(),
-                      sender: getNameFromJID(stanza.getAttribute('from')!),
+                  case StanzaMessageType.LIVE_MESSAGE:
+                    const liveMessage: ReceivedMessageType = {
                       content:
                         stanza.getElementsByTagName('body')[0].textContent!,
+                      sender: getNameFromJID(stanza.getAttribute('from')!),
+                      sentAt: DateTime.now(),
                     };
-                    useMessagesState.getState().addMessage(msg);
+                    useChatItemState.getState().addMessage(liveMessage);
                     break;
                   case StanzaMessageType.HISTORY:
-                    const oldMsgStanza =
+                    const historyMsgStanza =
                       stanza.getElementsByTagName('message')[0];
-                    const oldMsg: ChatMessageType = {
-                      sentAt: getMsgDateTimeFromStanza(stanza),
-                      sender: getNameFromJID(
-                        oldMsgStanza.getAttribute('from')!,
-                      ),
+                    const historyMessage: ReceivedMessageType = {
                       content:
-                        oldMsgStanza.getElementsByTagName('body')[0]
+                        historyMsgStanza.getElementsByTagName('body')[0]
                           .textContent!,
+                      sender: getNameFromJID(
+                        historyMsgStanza.getAttribute('from')!,
```

Comment:
can you remove this assertion ?

Reply:
I can't remove this assertion. However, the attribute's existence is already checked in `determineMessageType` function, used in the switch statement.